cmd/contour/serve.go

```diff
@@ -247,11 +258,13 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 					FieldLogger: log.WithField("context", "IngressProcessor"),
 				},
```

Comment:
should we be supporting this for `Ingress` too?

Reply:
I missed mentioning this in the design doc, but I've understood that there is no intention to support backend TLS for `Ingress`.  So I believe upstream TLS code is missing completely there.  This is mentioned in older design doc [design/tls-backend-verification.md](https://github.com/projectcontour/contour/blob/main/design/tls-backend-verification.md).

Reply:
Yes sorry, it is the server authentication that is missing: TLS as itself is supported using annotation `projectcontour.io/upstream-protocol.tls` on service like you say, but server authentication can be only expressed with `HTTPProxy` `spec.routes.services[].validation`. 

We can still add client authentication regardless of not authenticating the server, though it is unusual to have it this way.

Reply:
I've now added support for client auth for `Ingress` too.